lib/std/math/log.zig

```diff
@@ -31,9 +31,11 @@ pub fn log(comptime T: type, base: T, x: T) T {
         .ComptimeInt => {
             return @as(comptime_int, math.floor(math.ln(@as(f64, x)) / math.ln(float_base)));
         },
-        .Int => {
-            // TODO implement integer log without using float math
-            return @floatToInt(T, math.floor(math.ln(@intToFloat(f64, x)) / math.ln(float_base)));
+
+        // TODO implement integer log without using float math
+        .Int => |IntType| switch (IntType.signedness) {
+            .signed => return @compileError("log not defined for signed integers"),
```

Comment:
Mathematically, it is defined, and since we support complex numbers already, we could even return the result.

Maybe change the message to something like "log wouldn't be an integer" or "log of signed integers not implemented"?